flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -147,8 +150,8 @@
 	public static String[] getPluginParentFirstLoaderPatterns(Configuration config) {
 		String base = config.getString(PLUGIN_ALWAYS_PARENT_FIRST_LOADER_PATTERNS);
 		String append = config.getString(PLUGIN_ALWAYS_PARENT_FIRST_LOADER_PATTERNS_ADDITIONAL);
-
-		return ArrayUtils.concat(base.split(";"), append.split(";"));
+		final Splitter splitter = Splitter.on(';').omitEmptyStrings();
+		return Iterables.toArray(Iterables.concat(splitter.split(base), splitter.split(append)), String.class);
 	}
```

Comment:
Can you deduplicate this code with `getParentFirstLoaderPatterns`? Provide a base method `private static String[] getParentFirstLoaderPatterns(String base, String append)`? Especially that the original method that was copied/duplicated didn't have the bug.

Also maybe a unit test since this has failed/caused a bug?

Reply:
Added a test.

Reply:
And what about the deduplication?